### PR TITLE
Fixes the addressof_ref function of SyclContext and improves unit tests.

### LINKED ISSUE #484: AttributeError when calling addressof_ref in SyclContext
The attribute `_ctx_ref` was renamed to `_ctxt_ref` but the `addressof_ref` function was not updated. Calling `SyclContext.addressof_ref` causes and `AttributeError` for this reason:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "dpctl/_sycl_context.pyx", line 353, in dpctl._sycl_context.SyclContext.addressof_ref
AttributeError: 'dpctl._sycl_context.SyclContext' object has no attribute '_ctx_ref'

```

### PR DESCRIPTION
**Pull request by diptorupd**
Closes #484 

- Fixes a typo in the `addressof_ref` function of `SyclContext` that was causing an `AttributeError` and added a unit test for the member.
- Also, added missing unit tests for `SyclContext` and removed the duplicate tests that were copied over from `SyclDevice`.

### CODE REVIEW

File: dpctl/_sycl_context.pyx [hunk 1]
```diff
@@ -350,7 +350,7 @@ cdef class SyclContext(_SyclContext):
             used to create this :class:`dpctl.SyclContext` cast to a
             ``size_t``.
         """
-        return int(<size_t>self._ctx_ref)
+        return int(<size_t>self._ctxt_ref)
```
Comment (oleksandr-pavlyk) [inline dpctl/_sycl_context.pyx:353]:
```suggestion
        return <size_t>self._ctxt_ref
```

**Review by oleksandr-pavlyk — DISMISSED** [review]

File: dpctl/_sycl_context.pyx (hunk 1 above)
Comment (diptorupd) [reply dpctl/_sycl_context.pyx]: Done!

**Review by oleksandr-pavlyk — APPROVED** [review]
